### PR TITLE
Processing context variables in a fixed order

### LINKED ISSUE #498: Context variable replacement is ambiguous: GIANTSWARM_ENV(IRONMENT)
We support the use of certain predefined variables which are replces by the app-service with the according values on service creation.

A user reported a problem with this.

> if i try to use `$GIANTSWARM_ENVIRONMENT` in a swarm definition file i end up with a value like `mapleapps/stagingIRONMENT`, because it conflicts with `$GIANTSWARM_ENV`

Obviously the order in which we parse these variables is bad.

Original issue: https://github.com/giantswarm/user-support/issues/59#issuecomment-159631891

### PR DESCRIPTION
**Pull request by marians**
Fixes #498 

The context variable keys are sorted in reverse order so that GIANTSWARM_ENVIRONMENT is processed before GIANTSWARM_ENV.

### CODE REVIEW

**Comment by marians** [comment]
RFR @ewoutp @zeisss

**Comment by marians** [comment]
Here is a test service:

```
{
  "name": "busywait",
  "components": {
    "box": {
      "image": "busybox:latest",
      "args": [
        "sh",
        "-c",
        "while true; do echo 'short:$GIANTSWARM_ENVIRONMENT long:$GIANTSWARM_ENV'; sleep 10; done"
      ]
    }
  }
}
```

A `swarm create` and then `swarm cat` is enough to test.

File: create/utils.go [hunk 1]
```diff
@@ -208,8 +209,17 @@ func mergeVarMaps(srcs ...EnvMap) EnvMap {
 	dest := EnvMap{}
 	for i := len(srcs) - 1; i >= 0; i-- {
 		src := srcs[i]
-		for k, v := range src {
-			dest[k] = v
+		// Reverse sorting the keys is important
+		// so that GIANTSWARM_ENVIRONMENT is processed
+		// before GIANTSWARM_ENV!
+		var keys sort.StringSlice = make([]string, 0, len(src))
```
Comment (ewoutp) [inline create/utils.go]: This looks really complicated. You can probably reduce it to:

```
keys := []string{}
for key, _ := range src {
	keys = append(keys, key)
}
sort.Sort(sort.Reverse(sort.StringSlice(keys)))
```

Comment (marians) [inline create/utils.go]: Done

**Comment by marians** [comment]
@ewoutp Looking good to you now?

**Comment by ewoutp** [comment]
Does not build, needs #532.

**Comment by ewoutp** [comment]
I'm merged master, it builds now.

LGTM

**Pull request merged**